scripts/test-harness-contracts.ps1

```
@@ -881,6 +881,19 @@ try {
                 Check ((Invoke-Writegate $wg @{ command = 'npm test > $env:PELIZZAI_NAO_EXISTE_XYZ/f.log' } $wgTemp) -eq 0) "writegate does not block a target with an unresolvable variable ($leaf)"
                 # Non-regression: `>` inside quotes is text, not a redirect.
                 Check ((Invoke-Writegate $wg @{ command = 'git commit -m "a > b"' } $wgTemp) -eq 0) "writegate does not mistake quoted text for a redirect ($leaf)"
+
+                # -- Quote-aware SEGMENTATION (issue #74): a `|` inside quotes is text, not a pipe. --
+                # The raw split broke `sed -i 's|a|b|' pelizzai/...` mid-expression, elected a wrong
+                # target, and blocked the very carve-out the hook's message promises. Matrix from the
+                # issue: A must pass, B stays passing, C/D stay blocking, E (real pipe) stays passing.
+                Check ((Invoke-Writegate $wg @{ command = "sed -i -e 's|^- phase: exec|- phase: done|' pelizzai/data/state.md" } $wgTemp) -eq 0) "writegate: sed with pipe delimiter targeting pelizzai/ is allowed on a protected branch ($leaf)"
```

PSScriptAnalyzer (1.25.0)

[info] Cmdlet 'Invoke-Writegate' has positional parameter. Please use named parameters instead of positional parameters when calling a command.

(PSAvoidUsingPositionalParameters)